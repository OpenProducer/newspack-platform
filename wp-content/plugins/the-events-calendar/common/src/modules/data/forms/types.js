/**
 * Internal dependencies
 */
import { PREFIX_COMMON_STORE } from '@moderntribe/common/data/utils';

export const ADD_FORM = `${ PREFIX_COMMON_STORE }/ADD_FORM`;
export const SET_FORM_FIELDS = `${ PREFIX_COMMON_STORE }/SET_FORM_FIELDS`;
export const CREATE_FORM_DRAFT = `${ PREFIX_COMMON_STORE }/CREATE_FORM_DRAFT`;
export const EDIT_FORM_ENTRY = `${ PREFIX_COMMON_STORE }/EDIT_FORM_ENTRY`;
export const SUBMIT_FORM = `${ PREFIX_COMMON_STORE }/SUBMIT_FORM`;
export const CLEAR_FORM = `${ PREFIX_COMMON_STORE }/CLEAR_FORM`;
export const SET_SAVING_FORM = `${ PREFIX_COMMON_STORE }/SET_SAVING_FORM`;

export const ADD_VOLATILE_ID = `${ PREFIX_COMMON_STORE }/ADD_VOLATILE_ID`;
export const REMOVE_VOLATILE_ID = `${ PREFIX_COMMON_STORE }/REMOVE_VOLATILE_ID`;
