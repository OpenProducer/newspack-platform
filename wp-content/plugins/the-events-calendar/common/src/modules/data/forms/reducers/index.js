export { default as form } from './form';
export { default as volatile } from './volatile';
