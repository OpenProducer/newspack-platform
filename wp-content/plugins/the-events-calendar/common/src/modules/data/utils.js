export const PREFIX_COMMON_STORE = '@@MT/COMMON';
