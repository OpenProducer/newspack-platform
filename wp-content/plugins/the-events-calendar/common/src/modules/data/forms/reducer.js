/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './types';
import { form, volatile } from './reducers';

const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_FORM:
		case types.CLEAR_FORM:
		case types.SET_FORM_FIELDS:
		case types.CREATE_FORM_DRAFT:
		case types.EDIT_FORM_ENTRY:
		case types.SUBMIT_FORM:
		case types.SET_SAVING_FORM:
			return {
				...state,
				[ action.payload.id ]: form( state[ action.payload.id ], action ),
			};
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	volatile,
} );
