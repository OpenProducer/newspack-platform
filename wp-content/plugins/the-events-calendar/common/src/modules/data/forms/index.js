/**
 * Internal dependencies
 */
import reducer from './reducer';

import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';

export default reducer;

export { types, actions, selectors };
