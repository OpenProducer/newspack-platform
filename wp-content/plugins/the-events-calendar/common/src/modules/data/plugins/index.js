/**
 * Internal dependencies
 */
import reducer from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import * as constants from './constants';
import * as proptypes from './proptypes';

export default reducer;
export { types, actions, selectors, constants, proptypes };
