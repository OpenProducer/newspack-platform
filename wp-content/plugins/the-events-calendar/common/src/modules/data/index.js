/**
 * Internal Dependencies
 */
import * as editor from './editor/post-types';
import * as forms from './forms';
import * as plugins from './plugins';
export { default } from './reducers';

export { editor, forms, plugins };