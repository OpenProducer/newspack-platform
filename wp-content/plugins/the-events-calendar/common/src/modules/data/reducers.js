/**
 * External dependencies
 */
import { setupCreateReducer } from '@nfen/redux-reducer-injector';
import forms from './forms';

/**
 * Internal dependencies
 */
import plugins from './plugins';

export default setupCreateReducer( {
	plugins,
	forms,
} );
