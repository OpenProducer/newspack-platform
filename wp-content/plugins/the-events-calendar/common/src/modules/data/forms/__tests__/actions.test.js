/**
 * External dependencies
 */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/common/data/forms';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

describe( '[STORE] - Form actions', () => {
	test( 'Register form action', () => {
		expect( actions.registerForm( 20, 'tribe_organizers' ) ).toMatchSnapshot();
	} );

	test( 'Create draft entry', () => {
		expect( actions.createDraft( 20, { title: 'Modern Tribe' } ) ).toMatchSnapshot();
	} );

	test( 'Edit the entry action', () => {
		expect( actions.editEntry( 20, { title: 'Tribe' } ) ).toMatchSnapshot();
	} );

	test( 'Clear form action', () => {
		expect( actions.clearForm( 20 ) ).toMatchSnapshot();
	} );

	test( 'Set submit form', () => {
		expect( actions.setSubmit( 20 ) ).toMatchSnapshot();
	} );

	test( 'Set saving action', () => {
		expect( actions.setSaving( 20, true ) ).toMatchSnapshot();
		expect( actions.setSaving( 20, false ) ).toMatchSnapshot();
	} );

	test( 'Add volatile action', () => {
		expect( actions.addVolatile( 20 ) ).toMatchSnapshot();
	} );

	test( 'Remove volatile action', () => {
		expect( actions.removeVolatile( 20 ) ).toMatchSnapshot();
	} );
} );

describe( '[STORE] - form thunk actions', () => {
	let store = {};
	beforeAll( () => {
		store = mockStore( {
			events: {
			},
			forms: {
				byId: {
					20: {
						create: true,
						type: 'tribe_organizer',
						fields: {},
					},
					21: {
						create: false,
						type: 'tribe_venue',
						fields: {
							id: 21,
						},
					},
				},
				volatile: [],
			},
		} );
	} );

	afterEach( () => store.clearActions() );

	test( 'Send the form action when creating', () => {
		store.dispatch( actions.sendForm( 20, { title: 'Modern Tribe' } ) );
		expect( store.getActions() ).toMatchSnapshot();
	} );

	test( 'Send the form when editing', () => {
		store.dispatch( actions.sendForm( 21, { title: 'Tribe' } ) );
		expect( store.getActions() ).toMatchSnapshot();
	} );

	test( 'Maybe remove entry action without details', () => {
		store.dispatch( actions.maybeRemoveEntry( 20, {} ) );
		expect( store.getActions() ).toMatchSnapshot();
	} );

	test( 'Maybe remove entry action with details', () => {
		store.dispatch( actions.maybeRemoveEntry( 21, { id: 21, title: 'Modern Tribe' } ) );
		expect( store.getActions() ).toMatchSnapshot();
	} );
} );
