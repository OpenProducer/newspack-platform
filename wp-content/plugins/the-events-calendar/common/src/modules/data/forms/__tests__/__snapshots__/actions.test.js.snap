// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Form actions Add volatile action 1`] = `
Object {
  "payload": Object {
    "id": 20,
  },
  "type": "@@MT/COMMON/ADD_VOLATILE_ID",
}
`;

exports[`[STORE] - Form actions Clear form action 1`] = `
Object {
  "payload": Object {
    "id": 20,
  },
  "type": "@@MT/COMMON/CLEAR_FORM",
}
`;

exports[`[STORE] - Form actions Create draft entry 1`] = `
Object {
  "payload": Object {
    "fields": Object {
      "title": "Modern Tribe",
    },
    "id": 20,
  },
  "type": "@@MT/COMMON/CREATE_FORM_DRAFT",
}
`;

exports[`[STORE] - Form actions Edit the entry action 1`] = `
Object {
  "payload": Object {
    "fields": Object {
      "title": "Tribe",
    },
    "id": 20,
  },
  "type": "@@MT/COMMON/EDIT_FORM_ENTRY",
}
`;

exports[`[STORE] - Form actions Register form action 1`] = `
Object {
  "payload": Object {
    "id": 20,
    "type": "tribe_organizers",
  },
  "type": "@@MT/COMMON/ADD_FORM",
}
`;

exports[`[STORE] - Form actions Remove volatile action 1`] = `
Object {
  "payload": Object {
    "id": 20,
  },
  "type": "@@MT/COMMON/REMOVE_VOLATILE_ID",
}
`;

exports[`[STORE] - Form actions Set saving action 1`] = `
Object {
  "payload": Object {
    "id": 20,
    "saving": true,
  },
  "type": "@@MT/COMMON/SET_SAVING_FORM",
}
`;

exports[`[STORE] - Form actions Set saving action 2`] = `
Object {
  "payload": Object {
    "id": 20,
    "saving": false,
  },
  "type": "@@MT/COMMON/SET_SAVING_FORM",
}
`;

exports[`[STORE] - Form actions Set submit form 1`] = `
Object {
  "payload": Object {
    "id": 20,
  },
  "type": "@@MT/COMMON/SUBMIT_FORM",
}
`;

exports[`[STORE] - form thunk actions Maybe remove entry action with details 1`] = `
Array [
  Object {
    "meta": Object {
      "actions": Object {
        "success": [Function],
      },
      "path": "tribe_venue/21",
    },
    "type": "@@MT/COMMON/WP_REQUEST",
  },
]
`;

exports[`[STORE] - form thunk actions Maybe remove entry action without details 1`] = `Array []`;

exports[`[STORE] - form thunk actions Send the form action when creating 1`] = `
Array [
  Object {
    "meta": Object {
      "actions": Object {
        "error": [Function],
        "start": [Function],
        "success": [Function],
      },
      "params": Object {
        "body": "{\\"title\\":\\"Modern Tribe\\"}",
        "method": "POST",
      },
      "path": "tribe_organizer",
    },
    "type": "@@MT/COMMON/WP_REQUEST",
  },
]
`;

exports[`[STORE] - form thunk actions Send the form when editing 1`] = `
Array [
  Object {
    "meta": Object {
      "actions": Object {
        "error": [Function],
        "start": [Function],
        "success": [Function],
      },
      "params": Object {
        "body": "{\\"title\\":\\"Tribe\\"}",
        "method": "PUT",
      },
      "path": "tribe_venue/21",
    },
    "type": "@@MT/COMMON/WP_REQUEST",
  },
]
`;
